### DISCUSSION TITLE
Does Hardhat+Viem really replace Typechain?

### DISCUSSION THREAD

**Discussion by aalmada** [ethereum-ts/TypeChain#897] [General]
I've returned to working with Solidity+TypeScript after taking a break for a couple of years. As I dive back in, I'm trying to catch up on what's changed. One thing that caught my attention is the soft deprecation notice for Typechain. I'm currently experimenting with Hardhat+Viem, which is mentioned in the notice. According to the linked page, `hardhat-viem` "generates types for your contracts, enhancing type-checking and IDE suggestions." However, I haven't experienced any contract method name suggestions or type error warnings in VS Code.

While this doesn't seem to be a Typechain-specific issue, I'm mentioning it here because I didn't encounter these problems with Typechain before, and the notice suggests giving Hardhat+Viem a try.

Another feature I miss from Typechain is the ability to use its generated interfaces as types in my frontend code, as I demonstrate in these articles: [link](https://aalmada.github.io/tags/web3/).

Am I overlooking something? Should I continue using Typechain? I'd appreciate your opinion.

**Comment by krzkaczor** [comment]
@aalmada questions regarding hardhat-viem should be directed to hardhat support. I tested it some time ago and everything worked as expected. 

>Another feature I miss from Typechain is the ability to use its generated interfaces as types in my frontend code, as I demonstrate in these articles: [link](https://aalmada.github.io/tags/web3/).

I personally use viem + wagmi on frontend as well, so there is no need for TypeChain's generated interfaces. 

That being said, feel free to continue using TypeChain, it works.

**Reply by aalmada** [reply]
Thank you so much for the reply and all the work you did for Typechain!
I had already opened a discussion on hardhat: https://github.com/NomicFoundation/hardhat/discussions/5031
